src/main/kotlin/no/nav/familie/ba/sak/kjerne/behandling/behandlingstema/BehandlingstemaService.kt

```diff
@@ -25,137 +28,155 @@ class BehandlingstemaService(
     private val oppgaveService: OppgaveService,
     private val vilkårsvurderingTidslinjeService: VilkårsvurderingTidslinjeService,
     private val vilkårsvurderingRepository: VilkårsvurderingRepository,
+    private val clockProvider: ClockProvider,
 ) {
     @Transactional
-    fun oppdaterBehandlingstema(
+    fun oppdaterBehandlingstemaForRegistrereSøknad(
+        behandling: Behandling,
+        nyUnderkategori: BehandlingUnderkategori,
+    ): Behandling {
+        if (behandling.skalBehandlesAutomatisk) {
+            return behandling
+        }
+        return oppdaterBehandlingstemaPåBehandlingHvisNødvendig(behandling, behandling.kategori, nyUnderkategori)
+    }
+
+    @Transactional
+    fun oppdaterSaksbehandletBehandlingstema(
+        behandling: Behandling,
+        nyKategori: BehandlingKategori,
+        nyUnderkategori: BehandlingUnderkategori,
+    ): Behandling {
+        if (behandling.skalBehandlesAutomatisk) {
+            throw FunksjonellFeil("Kan ikke oppdatere behandlingstema på behandlinger som skal behandles automatisk.")
+        }
+        val forrigeKategori = behandling.kategori
+        val forrigeUnderkategori = behandling.underkategori
+        val oppdatertBehanding = oppdaterBehandlingstemaPåBehandlingHvisNødvendig(behandling, nyKategori, nyUnderkategori)
+        loggService.opprettEndretBehandlingstema(
+            behandling = oppdatertBehanding,
+            forrigeKategori = forrigeKategori,
+            forrigeUnderkategori = forrigeUnderkategori,
+            nyKategori = nyKategori,
+            nyUnderkategori = nyUnderkategori,
+        )
+        return oppdatertBehanding
+    }
+
+    @Transactional
+    fun oppdaterBehandlingstemaForVilkår(
         behandling: Behandling,
-        overstyrtKategori: BehandlingKategori? = null,
         overstyrtUnderkategori: BehandlingUnderkategori? = null,
-        manueltOppdatert: Boolean = false,
     ): Behandling {
-        if (behandling.skalBehandlesAutomatisk) return behandling
-        if (manueltOppdatert && (overstyrtKategori == null || overstyrtUnderkategori == null)) {
-            throw FunksjonellFeil("Du må velge behandlingstema.")
+        if (behandling.skalBehandlesAutomatisk) {
+            return behandling
         }
+        val nyKategori = finnKategori(behandling.fagsak.id)
+        val nyUnderkategori = overstyrtUnderkategori ?: finnUnderkategoriFraInneværendeBehandling(fagsakId = behandling.fagsak.id)
+        return oppdaterBehandlingstemaPåBehandlingHvisNødvendig(behandling, nyKategori, nyUnderkategori)
+    }
 
-        val utledetKategori =
-            bestemKategori(
-                overstyrtKategori = overstyrtKategori,
-                kategoriFraSisteIverksattBehandling = hentLøpendeKategori(behandling.fagsak.id),
-                kategoriFraInneværendeBehandling = hentKategoriFraInneværendeBehandling(behandling.fagsak.id),
-            )
+    fun finnKategori(fagsakId: Long): BehandlingKategori {
```

Comment:
```suggestion
    fun finnBehandlingKategori(fagsakId: Long): BehandlingKategori {
```
Tror det er greit å få på spesifikt behandlingkategori her, siden det også finnes underkategori (unngå forvirrelser om det er BEHANDLINGkategori eller UNDERkategori de ter snakk om)